util/top_recommendations.py

```diff
@@ -0,0 +1,39 @@
+import bisect
+
+
+class TopRecommendations(object):
+
+    def __init__(self, n_recommendations):
+        self.recommendations_values = []
+        self.recommendations_indices = []
+        self.n_recommendations = n_recommendations
+
+    def insert(self, index, value):
+        if self.n_recommendations > len(self.recommendations_values):
+            inserted_index = self._insert_and_return_index(self.recommendations_values, value)
+            self.recommendations_indices = self._insert_at_index(self.recommendations_indices, index, inserted_index)
+            return
+        if len(self.recommendations_values) != 0:
```

Comment:
if you wanted to break the function at the previous line, you can use 'elif' instead of 'if'.
if self.recommendations_values: # should work